cli/replicasets/edit_topology.go

```diff
@@ -0,0 +1,235 @@
+	appendBoolOpt(&optsStrings, "all_rw", opts.AllRW)
+	appendFloatOpt(&optsStrings, "weight", opts.Weight)
+	appendStringOpt(&optsStrings, "vshard_group", opts.VshardGroup)
+
+	appendJoinServersOpt(&optsStrings, "join_servers", opts.JoinInstancesURIs)
+	appendStringsSliceOpt(&optsStrings, "failover_priority", opts.FailoverPriorityUUIDs)
+
+	return strings.Join(optsStrings, ", ")
+}
+
+func serializeEditInstancesOpts(opts *EditInstancesOpts) string {
+	instancesString := make([]string, len(*opts))
+
+	for i, editInstanceOpts := range *opts {
+		instancesString[i] = serializeEditInstanceOpts(editInstanceOpts)
+	}
+
+	return strings.Join(instancesString, ", ")
+}
+
+func serializeEditInstanceOpts(opts *EditInstanceOpts) string {
+	var optsStrings []string
+
+	appendStringOpt(&optsStrings, "uuid", &opts.InstanceUUID)
+	appendBoolOpt(&optsStrings, "expelled", &opts.Expelled)
+
+	return fmt.Sprintf("{ %s }", strings.Join(optsStrings, ", "))
+}
+
+func appendStringOpt(optsStrings *[]string, optName string, optValue *string) {
+	if optValue == nil || *optValue == "" {
+		return
+	}
+
+	optString := fmt.Sprintf("%s = '%s'", optName, *optValue)
+	*optsStrings = append(*optsStrings, optString)
+}
+
+func appendBoolOpt(optsStrings *[]string, optName string, optValue *bool) {
+	if optValue == nil {
+		return
+	}
+
+	optString := fmt.Sprintf("%s = %t", optName, *optValue)
+	*optsStrings = append(*optsStrings, optString)
+}
+
+func appendFloatOpt(optsStrings *[]string, optName string, optValue *float64) {
+	if optValue == nil {
+		return
+	}
+
+	formattedOpt := strconv.FormatFloat(*optValue, 'f', -1, 64)
+
+	optString := fmt.Sprintf("%s = %s", optName, formattedOpt)
+	*optsStrings = append(*optsStrings, optString)
+}
+
+func appendStringsSliceOpt(optsStrings *[]string, optName string, optValue []string) {
+	if optValue == nil {
+		return
+	}
+
+	optString := fmt.Sprintf("%s = %s", optName, serializeStringsSlice(optValue))
+	*optsStrings = append(*optsStrings, optString)
+}
+
+func appendJoinServersOpt(optsStrings *[]string, optName string, instancesURIs []string) {
+	if len(instancesURIs) == 0 {
+		return
+	}
+
+	joinServerStrings := make([]string, len(instancesURIs))
+	for i, instancesURI := range instancesURIs {
+		joinServerStrings[i] = fmt.Sprintf("{ uri = '%s' }", instancesURI)
+	}
+
+	optString := fmt.Sprintf("%s = { %s }", optName, strings.Join(joinServerStrings, ", "))
+	*optsStrings = append(*optsStrings, optString)
+}
+
+func serializeStringsSlice(stringsSlice []string) string {
+	elemStrings := make([]string, len(stringsSlice))
+	for i, elem := range stringsSlice {
+		elemStrings[i] = fmt.Sprintf("'%s'", elem)
+	}
+
+	return fmt.Sprintf("{ %s }", strings.Join(elemStrings, ", "))
+}
+
+var (
+	editReplicasetBodyTemplate = `
+local cartridge = require('cartridge')
+
+{{ .FormatTopologyReplicasetFunc }}
+
+local res, err = cartridge.admin_edit_topology({
+	replicasets = {
+		{ {{ .ReplicasetInput }} },
+	}
+})
+
+if err ~= nil then
+	return nil, err
```

Comment:
This error is an object from "errors" module. Do you consider it?

Reply:
Yes, `tostring(err)` is called for every error before returning it from `EvalTarantoolConn`.